test/html/scrollToElement.html

```diff
@@ -0,0 +1,155 @@
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+    <input id="fourth" type="text">
+    <p id="fourth-clue"></p>
+    <a id="fourth-a" href="#" onclick="scrollFunction('first');return false;">Click me to scroll to first element</a>
+    .<br />.<br />.<br />.<br />.<br />.<br />.<br />.<br />
+
+    <script>
+      function scrollFunction(id) {
+        let elmnt = document.getElementById(id);
+        let elmnt_a_tag = document.getElementById(id + "-a").focus();
+        let clue = document.getElementById(id + "-clue");
+        elmnt.scrollIntoView();
+      }
+
+      function elementsInViewPort() {
+        let first = document.querySelector("#first");
+        let second = document.querySelector("#second");
+        let third = document.querySelector("#third");
+        let fourth = document.querySelector("#fourth");
+
+        // Gives a clue if element is in viewport
+        if (inViewport(first)) {
+          let clue = document.querySelector("#first-clue");
+          clue.innerHTML = "In View";
+        } else {
+          let clue = document.querySelector("#first-clue");
+          clue.innerHTML = "Out of View";
+        }
+
+        // Gives a clue if element is in viewport
+        if (inViewport(second)) {
+          let clue = document.querySelector("#second-clue");
+          clue.innerHTML = "In View";
+        } else {
+          let clue = document.querySelector("#second-clue");
+          clue.innerHTML = "Out of View";
+        }
+
+        // Gives a clue if element is in viewport
+        if (inViewport(third)) {
+          let clue = document.querySelector("#third-clue");
+          clue.innerHTML = "In View";
+        } else {
+          let clue = document.querySelector("#third-clue");
+          clue.innerHTML = "Out of View";
+        }
+
+        // Gives a clue if element is in viewport
+        if (inViewport(fourth)) {
+          let clue = document.querySelector("#fourth-clue");
+          clue.innerHTML = "In View";
+        } else {
+          let clue = document.querySelector("#fourth-clue");
+          clue.innerHTML = "Out of View";
+        }
+
+      }
+
+      function inViewport(element) {
+        // Get the elements position relative to the viewport
+
+        var bb = element.getBoundingClientRect();
```

Comment:
This should be `const bb = ...` just for consistency with the above variable definitions.